T24374 hack reset

dnohales opened the pull request:
## Don't merge this yet

This is for early review, I still need to investigate the debian packaging part of this.

https://phabricator.endlessm.com/T24374

ramcq commented on eos-tech-support/eos-hack-reset-first-contact:
```diff
@@ -0,0 +1,42 @@
+    user_id=$(echo $user_row | cut -d : -f 3)
+
+    if [[ $user_id -ge 1000 && $user_id -le 1999 ]]; then
+        delete_tries=1
```
What is the thinking behind the retry loop? Under what circumstances are you expecting userdel to fail, and later succeed?

dnohales replied:
When you isolate `eos-hack-reset-first-contact.target`, the processes used by the users take some time to finish.

ramcq replied:
Could you achieve similar with After=systemd-kill-everything? (Whatever it's called.)

ramcq commented on eos-tech-support/eos-hack-reset-first-contact:
```diff
@@ -0,0 +1,42 @@
+        delete_tries=1
+        while true; do
+            echo "Deleting user $user ($user_id) (try: $delete_tries)"
+            userdel $user
```
Is it intentional that you are removing the user but not the actual home directory? This could have very weird/surprising effects - if you recreate user later with the same name, but they have a different UID, they will not own their home directory and then their user account will likely be unusable.

wjt replied:
Our FAQ for deleting all users uses `userdel -r` https://support.endlessm.com/hc/en-us/articles/360006827311-How-do-I-delete-all-users-from-my-computer-

dnohales replied:
I see, well, I wanted to be not that destructive, most of the time the new user, if you are going to use the same name, is gonna match the ID if that was your only user in the system. But you are right, let's clean it properly.

ramcq commented on eos-tech-support/eos-hack-reboot-to-first-contact:
```diff
@@ -0,0 +1,57 @@
+    cat <<EOF
+Usage: $0
+Reboot the machine and deletes all users in the system,
+so you can execute the first contact experience from
```
I would just name this something like eos-user-reset or eos-erase-users similar, as it's not specific to Hack and could be a useful CS tool generally.

ramcq commented on eos-tech-support/eos-hack-reset-quest:
```diff
@@ -0,0 +1,60 @@
+EOF
+}
+
+args=$(getopt -o h -l "help" -n "eos-hack-reset-quests" -- "$@")
```
Nitpick: this -n is inconsistent with the binary name.

ramcq replied:
($0 will always match!)

wjt replied:
`$(dirname "$0")` is even better!

wjt commented on eos-tech-support/eos-hack-reset-quest:
```diff
@@ -0,0 +1,60 @@
+    fi
+
+    flatpak run --command=jq --filesystem=~/.var/app/com.endlessm.GameStateService \
+        com.endlessm.HackToolbox \
```
This is weird: run app A, giving it access to app B's state, and manually modify the config file. This ties the file format of GameStateService (maintained outside the OS) to this script (maintained in the OS) so you're committing to supporting this file format forever. I think it would be better to ship this tool as part of GameStateService and just have this OS-level script run `flatpak run --command=eos-hack-reset com.endlessm.GameStateService "$@"`.

dnohales replied:
I just wanted to use `jq` :(

wjt commented on eos-tech-support/eos-hack-reboot-to-first-contact:
```diff
@@ -0,0 +1,57 @@
+fi
+
+echo "This script will reboot the system and DELETE ALL USERS so you can"
+echo "restart the Hack first contact experience and create a new user."
```
It may be worth noting here that this script will dump you at the Endless logo for a potentially long period of time while the script runs with no user-visible feedback.

dnohales commented:
Refactored the user deletion scripts so they are not Hack specific. Now they are named `eos-factory-reset*`. I structured these scripts so is feasible to add more "reset stages" like cleaning `/etc` or deleting/reinstalling apps in the future.

Added more flags so the user can select which stage to reset and a flag to disable user confirmation (so we can add a GUI for factory reset in the future).

Also, I moved the systemd files here: https://github.com/endlessm/eos-boot-helper/pull/247 . And started thinking that the scripts should be moved there as well for simplicity.

ramcq commented:
I agree both sides of the feature should live in the same repo (ie: over there)

ramcq commented:
Oops I didn't mean to hit close.

dnohales commented on eos-factory-reset/eos-factory-reset:
```diff
@@ -0,0 +1,93 @@
+            break
+            ;;
+    esac
+done
```
I hate shell scripting, I've chosen it because there were already many of them in the repo.

wjt replied:
I fully support the slogan “No More Non-Trivial Shell Scripts”

dnohales commented:
Gonna move this script to `hack-game-state-service`, closing.
